### DISCUSSION TITLE
Martin Jerry Tasmota Devices

### DISCUSSION THREAD

**Discussion by cavanaug** [RavenSystem/esp-homekit-devices#1961] [MEPLHAA Scripts Collection]
Since these all support Tasmota and thus have esp8266 can we work to get all of these supported in HAA with defined MEPLHAA scripts.    It would be great to be able to buy devices direct from Amazon that can be upgraded over the air to HAA with minimal effort.   It would be even more fantastic if the HAA manager could directly upgrade Tasmota devices to HAA and populate the script.   Wow, I think you could get tons of new users to HAA with something like that.   Plus it would be a great way to support companies that support the open source community by producing friendly hw.

The biggest question I have is if HAA can support the dimmers and 3-way dimmers, as Im not sure I understand how HAA manages dimmers like these.

Has anyone put HAA on these devices/dimmers??

**Comment by daryel** [comment]
I put HAA on a Martin Jerry outlet.  It's functional except the button, no idea why the button isn't working. When it's pressed there is no output to the logs either. 

GPIO: 
12: Relay
13: Button
4: LED

```
{
	"c": {
		"io": [
			[
				[12, 4], 2
			],
			[
				[13], 6
			]
		],
		"o": 4
	},
	"a": [{
		"t": 2,
		"0": {
			"r": [
				[12, 0],
				[4, 1]
			]
		},
		"1": {
			"r": [
				[12, 1],
				[4, 0]
			]
		},
		"b": [
			[13]
		],
		"s": 0
	}]
}
```

![image](https://github.com/RavenSystem/esp-homekit-devices/assets/2807094/09dc1b1e-24a9-45da-98e2-9a0b921e6471)

I have some Martin Jerry switches too, but not going to attempt converting them unless I can get a simple outlet to work first.

**Reply by RavenSystem** [reply]
```
{
	"c": {
		"io": [
			[
				[12, 4], 2
			],
			[
				[13], 6, 1
			]
		],
		"o": 4
	},
	"a": [{
		"t": 2,
		"0": {
			"r": [
				[12, 0],
				[4, 1]
			]
		},
		"1": {
			"r": [
				[12, 1],
				[4, 0]
			]
		},
		"b": [
			[13]
		],
		"s": 0
	}]
}
```

**Comment by daryel** [comment]
Awesome!  Worked perfectly.  I guess I was pretty close.  Flashing the rest of my devices as soon as I get something sent to your Paypal.  Much appreciated for the software!

**Reply by RavenSystem** [reply]
You were so close. Only enable internal pull-up resistor of button GPIO (13):
`[ [13], 6, 1 ]` at `"io":[  ]` array.
